libraries/AP_HAL_ChibiOS/Storage.cpp

```diff
@@ -44,64 +44,72 @@ extern const AP_HAL::HAL& hal;
 
 void Storage::_storage_open(void)
 {
-    if (_initialised) {
+    if (_initialisedType != StorageBackend::None) {
         return;
     }
 
-#ifdef USE_POSIX
-    // if we have failed filesystem init don't try again
-    if (log_fd == -1) {
-        return;
-    }
-#endif
-
     _dirty_mask.clearall();
 
 #if HAL_WITH_RAMTRON
-    using_fram = fram.init();
-    if (using_fram) {
-        if (!fram.read(0, _buffer, CH_STORAGE_SIZE)) {
+    if (fram.init()) {
+        if (fram.read(0, _buffer, CH_STORAGE_SIZE) == CH_STORAGE_SIZE) {
+            _save_backup();
+            _initialisedType = StorageBackend::FRAM;
+            hal.console->printf("Initialised Storage type=%d\n", _initialisedType);
```

Comment:
this happens so early that it isn't visible on the console as console isn't setup yet

Reply:
Changed back to using `::printf()`